core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -5654,21 +5657,165 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
                 switch (opcode) {
                     /* Memory */
                     case SIMD_v128_load:
+                    {
+                        uint32 offset, addr;
+                        offset =
+                            read_uint32(frame_ip); // TODO Check with an offset!
+                        addr = GET_OPERAND(uint32, I32, 0);
+                        frame_ip += 2;
+                        addr_ret = GET_OFFSET();
+                        CHECK_MEMORY_OVERFLOW(16);
+                        PUT_V128_TO_ADDR(frame_lp + addr_ret, LOAD_V128(maddr));
+                        break;
+                    }
                     case SIMD_v128_load8x8_s:
+#if ENABLE_NEON == 1
+                    {
+                        uint32 offset, addr;
+                        offset = read_uint32(frame_ip);
+                        addr = GET_OPERAND(uint32, I32, 0);
+                        frame_ip += 2;
+                        addr_ret = GET_OFFSET();
+                        CHECK_MEMORY_OVERFLOW(16);
+
+                        int8x8_t v8 = vld1_s8((const int8_t *)maddr);
+                        v8 = vrev64_s8(v8);
+                        int16x8_t v16 = vmovl_s8(v8);
+
+                        V128 result;
+                        vst1q_s16((int16_t *)&result, v16);
```

Comment:
I guess only this part is architecture specific so the other code can go outside of the `ifdef` (and possibly have an inefficient, plain C implementation in the `else` block?)

Reply:
The same for all the other opcodes below.